source/blog/2015-11-13-rspec-3-4-has-been-released.md

```diff
@@ -0,0 +1,414 @@
+take a very long time.
+
+In RSpec 3.4, the bisection algorithm is _much_ more intelligent. It now
+uses a recursive approach that is designed to minimize the number of
+attempts needed to identify the culprits. Early feedback on the new
+algorithm is quite positive: Sam Livingston-Gray reported that a [3.3 bisection
+ran all night](https://twitter.com/geeksam/status/656858995932573697) without
+completing, but with the new algorithm [it completed in only 20
+minutes!](https://twitter.com/geeksam/status/656949626495328256)
+
+Thanks to Simon Coffey for implementing this! If you'd like to learn
+more, I recommend you [check out his PR](https://github.com/rspec/rspec-core/pull/1997) --
+it contains some really useful diagrams explaining how the new algorithm works.
+
+### Core: Failure Output Improvements
+
+Good failure output has always been a priority for RSpec, but in 3.4
+it's much improved, in a few ways:
+
+#### Multi-line Code Snippets
+
+RSpec includes a code snippet from the expectation failure in the failure
+output. Before RSpec 3.4, this generally worked OK as long as your
+expectation fit on one line. If you formatted it across multiple lines, like
+this:
+
+~~~ ruby
+expect {
+  MyNamespace::MyClass.some_long_method_name(:with, :some, :arguments)
+}.to raise_error(/some error snippet/)
+~~~
+
+...then the failure would just print the first line (`expect {`)
+since that's the stackframe included in the exception. In RSpec 3.4,
+we now load [Ripper](http://ruby-doc.org/stdlib-2.2.0/libdoc/ripper/rdoc/Ripper.html)
+from the standard library if it's available in order to parse the
+source and determine how many lines to include for the full expectation
+expression. For a case like the snippet above, the failure output
+will now include the entire expression.
+
+Thanks to Yuji Nakayama for [implementing](https://github.com/rspec/rspec-core/pull/2083) this!
+
+#### Install `coderay` for Syntax Highlighting
+
+Taking this a step further, if the [`coderay`](http://coderay.rubychan.de/)
+gem is available, RSpec 3.4 will use it to syntax highlight the code snippet
+in your terminal. Here's an example of how that looks, using the code snippet
+from above:
+
+![Failure with syntax highlighting](/images/blog/multiline_failure_with_syntax_highlighting.png)
+
+#### Better Failure Source Detection
+
+RSpec finds the failure code snippet by looking through the exception
+stack trace for an appropriate frame. We could just use the top stack
+frame but that's generally not what you want: when you have an expectation
+failure, the top frame refers to a link in RSpec where the
+`RSpec::Expectations::ExpectationNotMetError` was raised, and you want
+to see the snippet from your `expect` call site rather than seeing
+a snippet of RSpec's code. Before RSpec 3.4, our solution for this
+was fairly naive: we just looked for the first
+stack frame from your spec file. In some situations this would display
```

Comment:
Good point.  I've edited.  Let me know what you think.

BTW, I also added a note about the `config.max_displayed_failure_line_count` option -- thoughts?